### PR TITLE
chore(CPlusPlus): add watering plants leetcode

### PR DESCRIPTION
**Pull request by VishnuBhaarath**
#640 Have added the solution in c++

### CODE REVIEW

File: algorithms/CPlusPlus/Arrays/watering-plants-leetcode.cpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#include<iostream>
+#include<bits/stdc++.h>
+using namespace std;
+
```
Comment (aayushjain7) [inline algorithms/CPlusPlus/Arrays/watering-plants-leetcode.cpp:4]: Provide some description for the program

File: algorithms/CPlusPlus/Arrays/watering-plants-leetcode.cpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      }
+        return steps;
+
+}
+
+
+
+
+
+int main(){
```
Comment (aayushjain7) [inline algorithms/CPlusPlus/Arrays/watering-plants-leetcode.cpp:28]: remove unnecessary space.

File: algorithms/CPlusPlus/Arrays/watering-plants-leetcode.cpp [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	vector<int> a{2,2,3,3};
+	int capacity=5;
+	int number_of_steps=wateringPlants(a,capacity);
+	cout<<"Number of steps needed ";
+	cout<<number_of_steps;
+	cout<<"\n";
+	return 0;
+}
```
Comment (aayushjain7) [inline algorithms/CPlusPlus/Arrays/watering-plants-leetcode.cpp:36]: provide examples and time compexity

**Review by aayushjain7 — CHANGES_REQUESTED** [review]

File: algorithms/CPlusPlus/Arrays/watering-plants-leetcode.cpp (hunk 1 above)
Comment (VishnuBhaarath) [reply algorithms/CPlusPlus/Arrays/watering-plants-leetcode.cpp:14]: Have added the problem statement

File: algorithms/CPlusPlus/Arrays/watering-plants-leetcode.cpp (hunk 2 above)
Comment (VishnuBhaarath) [reply algorithms/CPlusPlus/Arrays/watering-plants-leetcode.cpp]: Have removed it

File: algorithms/CPlusPlus/Arrays/watering-plants-leetcode.cpp (hunk 3 above)
Comment (VishnuBhaarath) [reply algorithms/CPlusPlus/Arrays/watering-plants-leetcode.cpp:40]: Time complexity and examples added.

**Review by aayushjain7 — CHANGES_REQUESTED** [review]
Also update the readme.MD file for this program.

**Review by aayushjain7 — APPROVED** [review]

**Comment by ming-tsai** [comment]
Hi @VishnuBhaarath,
Please check this github action error
image

**Comment by ming-tsai** [comment]
Hi @VishnuBhaarath,
Any update?

**Comment by ming-tsai** [comment]
Hi @VishnuBhaarath,
This pull request will be closed, when you update please re-open this pull request.

**Pull request closed without merging**